### ISSUE TITLE
VbaProjectPart generates `vbaProject.bin` file with incorrect file extension

### ISSUE THREAD

**Issue by jozefizso** [dotnet/Open-XML-SDK#1292]
**Describe the bug**
We are generating macro files using OpenXML SDK and when we add the `VbaProjectPart` to the document, we expect it to create a file named `vbaProject.bin` in the same way the Microsoft Office applications do.

OpenXML SDK will generate a `vbaProject.xml` file. Such VBA macro is functioning normally in Office applications.

**Screenshots**

![image](https://user-images.githubusercontent.com/287778/211084126-2422f318-e1d9-4cf4-b4b5-7ac9940e4e42.png)

![image](https://user-images.githubusercontent.com/287778/211084286-7dfef7d5-0806-480b-961b-adc651ea26ec.png)

**To Reproduce**

```cs
var vbaProjectPath = @"C:\dev\project\bin\vbaProject.bin";
var macroTemplate = PresentationDocument.CreateFromTemplate("MacroTemplate.potm");
var mainDoc = macroTemplate.PresentationPart;
if (mainDoc != null)
{
    var vbaProject = mainDoc.AddNewPart<VbaProjectPart>();
    using var reader = File.OpenRead(vbaProjectPath);
    vbaProject.FeedData(reader);
    reader.Close();
}
```

Sample repository: https://github.com/NetOfficeFw/vbamc/blob/macro/src/vbamc/VbaCompiler.cs#L101

**Observed behavior**
Adding the `VbaProjectPart` into any of these document types: **WordprocessingDocument**, **SpreadsheetDocument** or **PresentationDocument** will create a respective `vbaProject.xml` file.

The added VBA project is functional in Office applications despite having incorrect extension.

**Expected behavior**
The default extension for the `VbaProjectPart` should be `.bin` instead of `.xml` to make the filenames in line with Microsoft Office application behavior.

**Desktop:**
 - OS: Windows 10
 - Office version 365 build 2211
 - .NET Target: .NET 6
 - DocumentFormat.OpenXml Version: 2.16.0 and newer

**Additional context**

The `VbaProjectPart` part is defined here: https://github.com/OfficeDev/Open-XML-SDK/blob/main/test/DocumentFormat.OpenXml.Packaging.Tests/data/PartConstraintData.json

```json
{
  "Name": "DocumentFormat.OpenXml.Packaging.VbaProjectPart",
  "ContentType": "application/vnd.ms-office.vbaProject",
  "IsContentTypeFixed": true,
  "RelationshipType": "http://schemas.microsoft.com/office/2006/relationships/vbaProject",
  "TargetFileExtension": ".xml",
  "TargetName": "vbaProject",
  "TargetPath": ".",
  "Parts": [
    {
      "RelationshipType": "http://schemas.microsoft.com/office/2006/relationships/wordVbaData",
      "PartClassName": "VbaDataPart",
      "PartContentType": "application/vnd.ms-word.vbaData+xml",
      "MinOccursIsNonZero": false,
      "MaxOccursGreatThanOne": false,
      "FileFormat": "Office2007"
    }
  ]
}
```

Incorrect exention definition: https://github.com/OfficeDev/Open-XML-SDK/blob/main/test/DocumentFormat.OpenXml.Packaging.Tests/data/PartConstraintData.json#L3596

**Comment by hunyu** [comment]
HI @jozefizso 

Thank you for reporting this issue to us.
I will bring this issue up during our next triage.

Hung-Chun Yu
Microsoft Open Specifications Support

**Comment by hunyu** [comment]
> HI @jozefizso
> 
> Thank you for reporting this issue to us. I will bring this issue up during our next triage.
> 
> Hung-Chun Yu Microsoft Open Specifications Support

Internal testing underway - https://github.com/OfficeDev/Open-XML-SDK/pull/1301

**Comment by Asbjoedt** [comment]
Hi

Isn't an extension of .xml better to have than a .bin?
Isn't it Excel who should change from generating .bin to generating .xml?

**Comment by jozefizso** [comment]
> Hi
> 
> Isn't an extension of .xml better to have than a .bin in the context of the Office Open XML specification? Isn't it Excel who should change from generating .bin to generating .xml?

I would expect the OpenXML library to create documents in the same way the Office apps do.

It's a binary file, why it should have an `.xml` extension? And there are trillions of documents with `vbaProject.bin` files.

**Comment by tomjebo** [comment]
@Asbjoedt the `vbaProjectPart` part is actually not XML. In [[MS-OFFMACRO] 2.2.1.2](https://learn.microsoft.com/en-us/openspecs/office_standards/ms-offmacro/b0a9a4f9-3dcb-4a43-8f30-a9a0534d4bc2) you can see that the vbaProject part is specified in [[MS-OVBA]](https://learn.microsoft.com/en-us/openspecs/office_file_formats/ms-ovba/575462ba-bf67-4190-9fac-c275523c75fc) which describes the compound file binary format ([[MS-CFB]](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-cfb/53989ce4-7b05-4f8d-829b-d08d6148375b)) structure of this part. 

Therefore, as @jozefizso mentioned, it should have a .bin extension. This was an oversight by the SDK's code generation engine earlier in the history of the project.

**Comment by Asbjoedt** [comment]
@tomjebo thx for your reply. I think I knew better, I just had to try out the comment anyway, so thank you for responding professionally.

**Comment by mikeebowen** [comment]
This issue is resolved by #1301
